pkg/resource/resource.go

```diff
@@ -425,3 +433,32 @@ func StableNAndSomeMore(n int, names []string) string {
 	sort.Strings(cpy)
 	return FirstNAndSomeMore(n, cpy)
 }
+
+// AsStruct converts the given object to a structpb.Struct for usage with gRPC
+// connections.
+// Copied from Crossplane's internal composite controller package
+// "github.com/crossplane/crossplane/internal/controller/apiextensions/composite"
+// Source code at:
```

Comment:
```suggestion
// Copied from:
```

Nit: This is probably sufficient context.